src/lib/git.ts

```diff
@@ -383,73 +383,81 @@ export const patchApply: CherrypicklikeFunction = async ({
 
     consoleLog(`patch: ${patch}`);
 
-    if (patch === '') {
-      throw new BackportError('empty patch error TODO');
+    if (patch.trim() === '' || !target.directories) {
+      throw new BackportError('Patch is empty. Nothing to apply.');
     }
 
-    consoleLog(`source?: ${target.sourceDirectory}`);
-    // TODO: check leading slash on sourceDirectory
-    const strippedPathPatch = patch.replaceAll(
-      new RegExp(`^([+-]{3} [ab])/${target.sourceDirectory}/`, 'gm'),
-      (_, start) => `${start}/`,
-    );
-
-    consoleLog(`stripped: ${strippedPathPatch}`);
+    consoleLog(`Generated patch:\n${patch}`);
+    consoleLog(`Source directory: ${target.sourceDirectory}`);
+    consoleLog(`Target directories: ${target.directories.join(', ')}`);
 
+    // Step 2: Apply the patch once per target directory (version)
     const results = await sequentially(
       target.directories!,
       async (targetDirectory) => {
+        // Rewrite patch paths: replace /<sourceDirectory>/ with /<targetDirectory>/ in content paths
+        const versionedPatch = patch.replaceAll(
+          new RegExp(
+            `^([+-]{3} [ab])(/content/.+?/)${target.sourceDirectory}/`,
+            'gm',
+          ),
+          (_, start, prefix) => `${start}${prefix}${targetDirectory}/`,
```

Comment:
Should we ensure targetDirectory has appropriate leading/trailing slashes? (i.e. no trailing slash)